portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,24 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Megan!</h1>");
+    ArrayList<String> hellos = new ArrayList<String>();
+    hellos.add("Hello Megan!");
+    hellos.add("Hello there!");
+    hellos.add("Hello!");
```

Comment:
A few things on collections:

1. Let's prefer to use ImmutableLists (immutable means they can't be modified after constructing) when it makes sense to. In general, immutable objects are preferred when possible because they're generally safer. You can use the com.google.common.collect version: https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableList.html. 

In general, let's try to use this library for all container-like data structures (lists, maps, etc). Guava in general (Google's public version of Java that contains extra libraries like this one) should be utilized when possible. See https://guava.dev/ to learn more.

2. Let's make this a constant, class variable instead of defining it inside of this function. So at the top of this class (after the definition but before this method), you can declare a constant and use the ImmutableList.of() method. So it should look something like: 

private static final ImmutableList<String> HELLO_MESSAGES = ImmutableList.of("Hello ...", "Hello...", ....);

The reason is that it's good to separate the logic from the constants that don't change. If you ever just need to add another "hello" message, you don't need to sift through the logic code to find this variable. Likewise, it makes the actual business logic of the doGet method a lot cleaner and concise.

Reply:
Thanks! I just committed these changes.